### PR TITLE
[WIP] Documentation

### PR DESCRIPTION
**Pull request by Pamplemousse**
Added misc documentation as I went through some portions of the code.

As usual, being much noober in `angr` than other main contributors, please peer-review thoroughly!

Marked as `WIP` as it is an ongoing process, but feel free to merge if it's reviewed and would bring value to the next release!

### CODE REVIEW

File: angr/analyses/decompiler/clinic.py [hunk 1]
```diff
@@ -18,6 +18,10 @@
 class Clinic(Analysis):
     """
     A Clinic deals with AILments.
+
+    :param angr.knowledge.Function func:
+    :param list<OptimizationPass> optimization_passes:
```
Comment (rhelmot) [inline angr/analyses/decompiler/clinic.py]: There is real syntax you can use here! https://www.python.org/dev/peps/pep-0484/

`List[OptimizationPass]`

Comment (rhelmot) [reply angr/analyses/decompiler/clinic.py]: + many similar below

File: angr/calling_conventions.py [hunk 2]
```diff
@@ -88,8 +88,9 @@ class SimRegArg(SimFunctionArgument):
     """
     Represents a function argument that has been passed in a register.
 
-    :ivar string reg_name:    The name of the represented register.
-    :ivar int size:           The size of the register, in number of bytes.
+    :param string reg_name:    The name of the represented register.
+    :param int size:           The size of the register, in number of bytes.
+    :param set<> alt_offsets:
```
Comment (rhelmot) [inline angr/calling_conventions.py]: dict, not set. it is in fact a dict mapping ints (sizes of stores in bytes) to ints (sub-register offsets at which the value of that size should be stored)

File: angr/calling_conventions.py [hunk 3]
```diff
@@ -141,8 +142,10 @@ class SimStackArg(SimFunctionArgument):
     """
     Represents a function argument that has been passed on the stack.
 
-    :var int stack_offset:    The position of the argument relative to the stack pointer after the function prelude.
-    :ivar int size:           The size of the argument, in number of bytes.
+    :param int stack_offset:    The position of the argument relative to the stack pointer after the function prelude.
+    :param int size:            The size of the argument, in number of bytes.
+    :param int stack_offset:    The position of the argument as an offset, expressed in number of machine-words, from the
```
Comment (rhelmot) [inline angr/calling_conventions.py]: two stack_offset lines?

File: angr/engines/light/data.py [hunk 4]
```diff
@@ -63,6 +63,18 @@ def try_unpack_const(expr):
...
+    """
+    Represents a symbolic integer that equals to the given offset added the symbolic register's value.
+
+    For example, `RegisterOffset(4, eax, 8)` represents a (symbolic) integer between 0 and 15, that equals to `eax + 8`
+    regardless of what value `eax` holds.
+    Note that it does not matter whether `eax + 8` points to a valid memory location or not.
+
+    :param bits:                     The size of the resulting value, in number of bits.
+    :param string reg:               The register's name from which the value will be used to compute the offset from.
+    :param int|claripy.BVS offset:   The value of the offset. A concrete offset is modeled as an <int>, while a
```
Comment (rhelmot) [inline angr/engines/light/data.py]: 1) the syntax is `a or b`
2) claripy.BVS isn't a type. you want claripy.ast.BV

Comment (Pamplemousse) [reply angr/engines/light/data.py]: 1. Not sure where you have seen this syntax documented, but it gives me odd results.

The doc parser seems to rely on spaces to identify tokens...

Used [`Union[int,claripy.BVS.offset]`](https://docs.python.org/3.7/library/typing.html#typing.Union).

**Pull request closed without merging**